go/simple/v2/ampli/ampli_test.go

```diff
@@ -0,0 +1,148 @@
+	identify := NewIdentify(6.4).SetOptionalArray([]string{"a", "b"})
+
+	userProperties := map[amplitude.IdentityOp]map[string]interface{}{}
+	userProperties[amplitude.IdentityOpSet] = map[string]interface{}{
+		"requiredNumber": 6.4,
+		"optionalArray":  []string{"a", "b"},
+	}
+
+	assert.Equal(t, userProperties, identify.toAmplitudeEvent().UserProperties)
+}
+
+func TestGroup_NewGroup(t *testing.T) {
+	group := NewGroup(true)
+
+	expectEvent := amplitude.Event{
+		EventType: amplitude.GroupIdentifyEventType,
+		GroupProperties: map[amplitude.IdentityOp]map[string]interface{}{
+			amplitude.IdentityOpSet: {"requiredBoolean": true},
+		},
+	}
+
+	assert.Equal(t, expectEvent, group.toAmplitudeEvent())
+}
+
+func TestGroup_SetOptionalString(t *testing.T) {
+	group := NewGroup(true)
+	group.SetOptionalString("optional-string")
+
+	expectEvent := amplitude.Event{
+		EventType: amplitude.GroupIdentifyEventType,
+		GroupProperties: map[amplitude.IdentityOp]map[string]interface{}{
+			amplitude.IdentityOpSet: {"requiredBoolean": true, "optionalString": "optional-string"},
+		},
+	}
+
+	assert.Equal(t, expectEvent, group.toAmplitudeEvent())
+}
+
+func TestEventWithAllProperties(t *testing.T) {
+	event := NewEventWithAllProperties([]string{"abc", "test"}, true, EventWithAllPropertiesRequiredEnumEnum1, 3, 16.4, "str")
+
+	eventProperties := map[string]interface{}{
+		"requiredArray":   []string{"abc", "test"},
+		"requiredBool":    true,
+		"requiredEnum":    EventWithAllPropertiesRequiredEnumEnum1,
+		"requiredInteger": 3,
+		"requiredNumber":  16.4,
+		"requiredString":  "str",
+	}
+	// Should not have either optionalString or optionalBool in EventProperties
+	assert.Equal(t, eventProperties, event.toAmplitudeEvent().EventProperties)
+
+	event.SetOptionalString("optional-string")
+	eventProperties["optionalString"] = "optional-string"
+	// Should not have optionalBool in EventProperties
+	assert.Equal(t, eventProperties, event.toAmplitudeEvent().EventProperties)
+	// log.Printf("TestEventWithAllProperties: \n\t Should NOT have optionalBool in EventProperties: \n\t %+v", event.toAmplitudeEvent().EventProperties)
+
+	event.SetOptionalBool(false)
+	eventProperties["optionalBool"] = false
+	assert.Equal(t, eventProperties, event.toAmplitudeEvent().EventProperties)
+	// log.Printf("TestEventWithAllProperties: \n\t Should have optionalBool in EventProperties: \n\t %+v", event.toAmplitudeEvent().EventProperties)
+}
+
+func TestAmpli_LoadWithEnvironment(t *testing.T) {
+	expectConfig := amplitude.NewConfig("test-development-api-key")
+
+	instance := Ampli{}
+	APIKey[EnvironmentDevelopment] = "test-development-api-key"
+	instance.Load(LoadOptions{
+		Environment: EnvironmentDevelopment,
+	})
+
+	// Amplitude.Client.Logger is an interface of which equality is determined based on memory address
+	// So here only check one of the default configurations, FlushQueueSize
+	assert.Equal(t, expectConfig.APIKey, instance.Client.GetConfig().APIKey)
```

Comment:
Compiler error: `instance.Client.GetConfig undefined (type amplitude.Client has no field or method GetConfig)`

Reply:
I import core Go SDK from local where I added `GetConfig()`

Reply:
Usually Go code doesn't use `Get` prefix for function/methods names.

Reply:
I referred to [this](https://go.dev/doc/effective_go#Getters) where use capitalized getter method when struct is private, and `Set` prefix is likely to be used. Here as we already have a `Config` struct, do not want to mix it up with `Client.Config()` getter function. 

Changed it to `Config()` if you don't recommend `Get` prefix here.